runtime/commonMain/src/kotlinx/serialization/SerializerResolving.kt

```diff
@@ -44,10 +46,10 @@ public inline fun <reified T> serializer(): KSerializer<T> {
  * ```
  *
  * This is a computation-heavy call, so it is recommended to cache the result.
- * [typeOf] API currently does not work on Kotlin/JS.
+ * [typeOf] API currently does not work with user-defined generic classes on Kotlin/JS.
  */
 @Suppress("UNCHECKED_CAST", "NO_REFLECTION_IN_CLASS_PATH", "UNSUPPORTED")
-@ImplicitReflectionSerializer
+@UseExperimental(ImplicitReflectionSerializer::class)
```

Comment:
Why `UseExperimental` instead of `ImplicitReflectionSerializer`?

Reply:
Because this method correctly works with generics and overcomes lack of reflection on Native and JS via our internal mechanisms